include/GafferImage/OpenImageIOReader.h

```diff
@@ -116,6 +116,8 @@ class GAFFERIMAGE_API OpenImageIOReader : public ImageNode
 
 	private :
 
+		std::shared_ptr<void> retrieveFile( const Gaffer::Context *context, bool holdForBlack = false ) const;
```

Comment:
Could we forward-declare `File` here as a nested class and then avoid all the `static_pointer_casts` later?